ddtrace/llmobs/utils.py

```diff
@@ -76,16 +80,31 @@ def __init__(self, messages: Union[List[Dict[str, str]], Dict[str, str], str]):
                 continue
             elif not isinstance(message, dict):
                 raise TypeError("messages must be a string, dictionary, or list of dictionaries.")
+
             content = message.get("content", "")
             role = message.get("role")
             if not isinstance(content, str):
                 raise TypeError("Message content must be a string.")
-            if not role:
-                self.messages.append(Message(content=content))
-                continue
-            if not isinstance(role, str):
-                raise TypeError("Message role must be a string, and one of .")
-            self.messages.append(Message(content=content, role=role))
+
+            msg_dict = Message(content=content)
+            if role:
+                if not isinstance(role, str):
+                    raise TypeError("Message role must be a string.")
+                msg_dict["role"] = role
+
+            tool_calls = message.get("tool_calls")
+            if tool_calls is not None:
+                if not isinstance(tool_calls, list):
+                    raise TypeError("tool_calls must be a list.")
+                msg_dict["tool_calls"] = tool_calls
```

Comment:
Let's construct ToolCall objects instead of passing it directly as tool_calls. Let's also make sure `name` and `arguments` are present and correct types, the other ones we can add if it's present